journalpump/senders/kafka.py

```diff
@@ -60,6 +60,10 @@ def _init_kafka(self):
     def send_messages(self, *, messages, cursor):
         if not self.kafka_producer:
             self._init_kafka()
+
+        if self.kafka_producer is None:
+            raise Exception("kafka producer could not be initialized")
```

Comment:
This check should be in the init function.

Reply:
I agree but see above: moving the check to `__init__` would give us RAII semantics which would make mypy happy which would make everything happy, but it fundamentally changes the structure of `journalpump` in a big way I wouldn't want to embark on without the initial typechecks and really solid testing.